quark/mdk-2.0.q

```diff
@@ -106,10 +106,28 @@ namespace mdk {
         Session session();
 
         @doc("""
-             Create a new Session and join it to a distributed trace.
+             Create a new Session and join it to an existing distributed sesion.
+
+             This should only ever be done once per an encoded context. That
+             means you should only use it for RPC or similar one-off calls. If
+             you received the encoded context via a broadcast medium (pub/sub,
+             message queues with multiple readers, etc.) you should use
+             childSession() instead.
              """)
         Session join(String encodedContext);
 
+        @doc("""
+             Create a new Session. All properties will of the given encoded
+             distributed session be inherited, e.g. timeouts and overrides.
+
+             This is intended for use for encoded context received via a
+             broadcast medium (pub/sub, message queues with multiple readers,
+             etc.). If you know only you received the encoded context,
+             e.g. you're coding a server that receives the context from a HTTP
+             request, you should join() instead.
+             """)
+        Session childSession(String encodedContext);
```

Comment:
I think the term childSession is confusing because it implies a tree structure, which is exactly the opposite of what we are trying to convey here (join is actually the one creating a tree structure).

Just brainstoring, but maybe Session.spawn(...), Session.fork(...), something like that?